### PR TITLE
Linux support

### PR DESCRIPTION
**Pull request by kkwpsv**
X11 window embedding now works. Tested on Debian 12 x64 and [Kylin V10](https://en.wikipedia.org/wiki/Kylin_(operating_system)) (base on Debian 11) arm64.

But there's still some issues on arm64.
If we load libcef.so dynamic, it will fail with "cannot allocate memory in static TLS block". Maybe CLR uses too much TLS?
I tried `glibc.rtld.optional_static_tls` . It doesn't work. Maybe my linux os not support it. I don't know how to investigate it further.

Then I tried `LD_PRELOAD`, it works. But we need to do additional things to make it work with Avalonia. Cef uses `HarfBuzz`, and Avalonia also uses it. We must set `LD_PRELOAD=/path/to/libHarfBuzzSharp.so:/path/to/libcef.so` to ensure libHarfBuzzSharp load before cef, or the process will crash with SIGSEGV in HarfBuzz call.

We can also use [patchelf](https://github.com/NixOS/patchelf) to modify the elf file to load cef when process loading.
patchelf --add-needed libHarfBuzzSharp.so --add-needed libcef.so path/to/Xilium.CefGlue.Demo.Avalonia
patchelf --add-needed libcef.so path/to/Xilium.CefGlue.BrowserProcess (Browser process doesn't use Avalonia, so there's no need to add libHarfBuzzSharp.so)

### CODE REVIEW

**Comment by walterlv** [comment]
@kkwpsv Thanks a lot for your great work on this! Although there are still some issues, at least it can now be used on production environments.

**Comment by lindexi** [comment]
Awesome!

File: CefGlue.Avalonia/Platform/Linux/XWindow.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+        private static void EnsureDisplay()
+        {
+            if (_display == IntPtr.Zero)
+            {
+                _display = XOpenDisplay(null);
+            }
+        }
+
+        public void Dispose()
```
Comment (lindexi) [inline CefGlue.Avalonia/Platform/Linux/XWindow.cs:54]: Should we check `_needDispose` here?

File: CefGlue.Avalonia/Platform/Linux/XWindow.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        {
+            EnsureDisplay();
+            XDestroyWindow(_display, Handle);
+            XFlush(_display);
```
Comment (lindexi) [inline CefGlue.Avalonia/Platform/Linux/XWindow.cs:58]: Should we dispose `_display` here?

**Comment by Haltroy** [comment]
Looks OK on GitHub, will test this and create PR on the main CefGlue repository as soon as possible.

**Comment by Haltroy** [comment]
Update: It worked. thank you so much @kkwpsv will create the PR right now.

**Pull request merged**